discord/sink.py

```diff
@@ -0,0 +1,775 @@
+        super().__init__(version_flag, rtcp_type, length)
+
+        fmt = ">I"
+        buf_size = struct.calcsize(fmt)
+        self.ssrc = struct.unpack(fmt, data[:buf_size])[0]
+        self._parse_report_and_extension(data[buf_size:])
+
+
+class RTCPSourceDescriptionPacket(RTCPPacket):
+    """Source Description packet which describes sources.
+
+    Read in detail here: https://www.freesoft.org/CIE/RFC/1889/23.htm
+
+    Extends :class:`RTCPPacket` and inherits its attributes.
+
+    Attributes
+    ----------
+    chunks: Sequence[:class:`RTCPSourceDescriptionChunk`]
+        Sequence of chunks that contain items.
+    """
+    __slots__ = RTCPPacket.__slots__ + ("chunks",)
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        self.chunks = []
+        for _ in range(self.rc):
+            chunk, offset = self._parse_chunk(data)
+            data = data[offset:]
+            self.chunks.append(chunk)
+
+    def _parse_chunk(self, data):
+        chunk = RTCPSourceDescriptionChunk(struct.unpack(">I", data), [])
+        i = 4
+        while True:
+            cname = struct.unpack_from(">B", buffer=data, offset=i)[0]
+            i += 1
+            if cname == 0:
+                break
+
+            length = struct.unpack_from(">B", buffer=data, offset=i)[0]
+            i += 1
+            description = ""
+            if length > 0:
+                description = struct.unpack_from(f">{length}s", buffer=data, offset=i)
+                i += length
+
+            chunk.items.append(RTCPSourceDescriptionItem(cname, description))
+
+        # Chunks are padded by 32-bit boundaries
+        if i % 4 != 0:
+            i += 4 - (i % 4)
+        return chunk, i
+
+
+class RTCPGoodbyePacket(RTCPPacket):
+    """A Goodbye packet indicating a number of SSRCs that are disconnected
+    and possibly providing a reason for the disconnect
+
+    Read in detail here: https://www.freesoft.org/CIE/RFC/1889/32.htm
+
+    Extends :class:`RTCPPacket` and inherits its attributes.
+
+    Attributes
+    ----------
+    ssrc_byes: Sequence[:class:`int`]
+        List of SSRCs that are disconnecting.
+    reason: :class:`bytes`
+        Reason for disconnect.
+    """
+    __slots__ = RTCPPacket.__slots__ + ("ssrc_byes", "reason")
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        if self.rc == 0:
+            buf_size = 0
+            self.ssrc_byes = []
+        else:
+            buf_size = self.rc*struct.calcsize("I")
+            self.ssrc_byes = struct.unpack(f">{self.rc}I", data[:buf_size])
+        reason_length = struct.unpack(">B", data[buf_size:buf_size+1])
+        self.reason = b"" if reason_length == 0 else struct.unpack(f">{reason_length}s", data[buf_size+1:])
```

Comment:
similar to previous things on struct performance and convention + a fix 

```suggestion
            buf_size = self.rc * 4
            self.ssrc_byes = struct.unpack_from(f"!{self.rc}I", data, 0)
        reason_length = struct.unpack_from("!B", data, buf_size)
        self.reason = b"" if reason_length == 0 else struct.unpack_from(f"!{reason_length}s", data, buf_size+1)
```